controllers/contributorController.js

```diff
@@ -130,29 +106,61 @@ const controller = {
         .map((item) => item.trim())
         .filter((item) => validSkills.includes(item));
 
-      const newContributor = await ContributorModel.create({
-        title,
-        project_id: project._id,
-        skills: skillsArr,
-        is_remote,
-        description,
-        commitment_level,
-        available_slots,
-        city,
-        remuneration,
-      });
+      data.skills = skillsArr;
+      data.project_id = req.projectID;
+
+      const newContributor = await ContributorModel.create(data);
 
       return res.status(201).json(newContributor);
     } catch (error) {
+      console.log(`error: ${error}`);
       return res.status(500).json({
         error: 'Failed to fetch data',
       });
     }
-
   },
 
   update: async (req, res) => {
+    let data = null;
+    try {
+      data = await validator.details.validateAsync(req.body);
+    } catch (error) {
+      console.log(error);
+      return res.status(400).json({
+        error: 'Invalid input',
+      });
+    }
+
+    const relations = await RelationshipModel.find({ contributor_id: req.contributorID });
+
+    const skillsArr = data.skills
+      .split(',')
+      .map((item) => item.trim())
+      .filter((item) => validSkills.includes(item));
+    data.skills = skillsArr;
+
+    // cannot change available slots below the # of accepted users
+    const acceptedRelations = relations.filter((relation) => relation.state === 'accepted');
+    if (data.available_slots < acceptedRelations.length) {
+      return res.status(400).json({
+        error: 'Invalid input',
+      });
+    }
 
+    // Find and update the document
+    try {
+      const updatedContributor = await ContributorModel.findOneAndUpdate(
+        { _id: req.contributorID },
+        data,
+        { new: true }
+      );
+      return res.status(200).json(updatedContributor);
```

Comment:
Should return? Maybe if going to go back to contributor view but shouldn't if.

Reply:
can remove the data later on when we work on FE. for now I generally will return data when successful, just to confirm that it works.